src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -132,8 +131,8 @@ public String getNonProxyHosts() {
     return this.nonProxyHosts;
   }
 
-  public Protocol getProxyProtocol() {
-    return this.proxyProtocol.equalsIgnoreCase("https") ? Protocol.HTTPS : Protocol.HTTP;
+  public HttpProtocol getProxyProtocol() {
```

Comment:
I did what you suggested, but did not remove all the places we use  `getProxyProtocol`. There are 2 references left where we eventually need to create a `com.amazonaws.Protocol`:
* In `net.snowflake.client.core.HttpUtil#setProxyForS3` 
* In `net.snowflake.client.core.HttpUtil#buildHttpClient`

Reply:
`setProxyForS3` is a specific method for AWS environment and we should move the implementation to some class (e.g. `S3HttpUtil`) in the `net.snowflake.client.jdbc.cloud.storage` package (but we should still keep the public method in `HttpUtil` as deprecated and calling the new method)

usage of `getProxyProtocol` in `buildHttpClient` when creating `SnowflakeMutableProxyRoutePlanner` could also be unnecessary - we can create new constructor in `SnowflakeMutableProxyRoutePlanner` that receives `HttpProtocol` not AWS `Protocol` (and making existing constructor deprecated, field `protocol` in `SnowflakeMutableProxyRoutePlanner` could also be changed to `HttpProtocol` since it is private)

Reply:
Let's copy this class from AWS in a separate PR and issue to not block this PR - we need to discuss internally how we want to mark such copied classes

Reply:
ok I reverted the commit in 6e10f57bfb25ca7206378469798acf45929ddbe6